pkg/federate/kubefed/federator.go

```diff
@@ -46,6 +49,7 @@ func New(kubeFedConfig *rest.Config, stopChan <-chan struct{}) (federate.Federat
 		clusterMap:    make(map[string]*rest.Config),
 		kubeFedClient: kubeFedClient,
 		stopChan:      stopChan,
+		scheme:        scheme.Scheme,
```

Comment:
I'm curious why you store the `scheme` as a field instead of inlining `scheme.Scheme`. The UTs set it to `runtime.NewScheme()` which `scheme.Scheme` equates to. Or am I missing something?

Reply:
IIRC all types are registered in this scheme through their respective init()

Reply:
All k8s types are. But, eg, submariner CRD types (ie Endpoint and Cluster) define their own [Scheme](https://github.com/submariner-io/submariner/blob/master/pkg/client/clientset/versioned/scheme/register.go). I ran into the issue of using the correct `Scheme` in the cluster code for watching `KubeFedCluster` - I had to use the `sigs.k8s.io/kubefed/pkg/client/generic/scheme.Scheme`. I suspect we'll run into the same issue here but don't worry about it right now - I'll find out soon enough when I start using this for the submariner CRDs 😄